### PR TITLE
python3Packages.yamllint: 1.31.0 -> 1.32.0, switch to github source, …

### PR DESCRIPTION
**Pull request by NickCao**
…fix license

Changelog: https://github.com/adrienverge/yamllint/blob/v1.32.0/CHANGELOG.rst

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 238983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.pytest-testinfra</li>
    <li>python311Packages.pytest-testinfra.dist</li>
  </ul>
</details>
<details>
  <summary>81 packages built:</summary>
  <ul>
    <li>ansible (python310Packages.ansible-core)</li>
    <li>ansible-later</li>
    <li>ansible-later.dist</li>
    <li>ansible-lint</li>
    <li>ansible-lint.dist</li>
    <li>ansible.dist (python310Packages.ansible-core.dist)</li>
    <li>ansible_2_13</li>
    <li>ansible_2_13.dist</li>
    <li>ansible_2_14</li>
    <li>ansible_2_14.dist</li>
    <li>chart-testing</li>
    <li>fdroidserver</li>
    <li>fdroidserver.dist</li>
    <li>kargo</li>
    <li>kargo.dist</li>
    <li>mozillavpn</li>
    <li>mozphab</li>
    <li>mozphab.dist</li>
    <li>napalm</li>
    <li>napalm.dist</li>
    <li>peering-manager</li>
    <li>python310Packages.ansible</li>
    <li>python310Packages.ansible-compat</li>
    <li>python310Packages.ansible-compat.dist</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-kernel.dist</li>
    <li>python310Packages.ansible-runner</li>
    <li>python310Packages.ansible-runner.dist</li>
    <li>python310Packages.ansible.dist</li>
    <li>python310Packages.glean-parser</li>
    <li>python310Packages.glean-parser.dist</li>
    <li>python310Packages.glean-sdk</li>
    <li>python310Packages.glean-sdk.dist</li>
    <li>python310Packages.junos-eznc</li>
    <li>python310Packages.junos-eznc.dist</li>
    <li>python310Packages.napalm</li>
    <li>python310Packages.napalm-hp-procurve</li>
    <li>python310Packages.napalm-hp-procurve.dist</li>
    <li>python310Packages.napalm.dist</li>
    <li>python310Packages.netmiko</li>
    <li>python310Packages.netmiko.dist</li>
    <li>python310Packages.ntc-templates</li>
    <li>python310Packages.ntc-templates.dist</li>
    <li>python310Packages.pytest-ansible</li>
    <li>python310Packages.pytest-ansible.dist</li>
    <li>python310Packages.pytest-testinfra</li>
    <li>python310Packages.pytest-testinfra.dist</li>
    <li>ttp (python310Packages.ttp)</li>
    <li>ttp.dist (python310Packages.ttp.dist)</li>
    <li>yamllint (python310Packages.yamllint)</li>
    <li>yamllint.dist (python310Packages.yamllint.dist)</li>
    <li>python311Packages.ansible</li>
    <li>python311Packages.ansible-compat</li>
    <li>python311Packages.ansible-compat.dist</li>
    <li>python311Packages.ansible-core</li>
    <li>python311Packages.ansible-core.dist</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-kernel.dist</li>
    <li>python311Packages.ansible-runner</li>
    <li>python311Packages.ansible-runner.dist</li>
    <li>python311Packages.ansible.dist</li>
    <li>python311Packages.glean-parser</li>
    <li>python311Packages.glean-parser.dist</li>
    <li>python311Packages.glean-sdk</li>
    <li>python311Packages.glean-sdk.dist</li>
    <li>python311Packages.junos-eznc</li>
    <li>python311Packages.junos-eznc.dist</li>
    <li>python311Packages.napalm</li>
    <li>python311Packages.napalm-hp-procurve</li>
    <li>python311Packages.napalm-hp-procurve.dist</li>
    <li>python311Packages.napalm.dist</li>
    <li>python311Packages.netmiko</li>
    <li>python311Packages.netmiko.dist</li>
    <li>python311Packages.ntc-templates</li>
    <li>python311Packages.ntc-templates.dist</li>
    <li>python311Packages.pytest-ansible</li>
    <li>python311Packages.pytest-ansible.dist</li>
    <li>python311Packages.ttp</li>
    <li>python311Packages.ttp.dist</li>
    <li>python311Packages.yamllint</li>
    <li>python311Packages.yamllint.dist</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/yamllint/default.nix [hunk 1]
```diff
@@ -45,7 +52,7 @@ buildPythonPackage rec {
     description = "A linter for YAML files";
     homepage = "https://github.com/adrienverge/yamllint";
     changelog = "https://github.com/adrienverge/yamllint/blob/v${version}/CHANGELOG.rst";
-    license = licenses.gpl3Plus;
+    license = licenses.gpl3Only;
```
Comment (figsoda) [inline pkgs/development/python-modules/yamllint/default.nix:55]: that doesn't look right - most files seems to be licensed gpl3+ according to the file headers, e.g. https://github.com/adrienverge/yamllint/blob/b05e028c5881819161d11cb543fd96a30c06cceb/yamllint/cli.py#L6

Comment (NickCao) [reply pkgs/development/python-modules/yamllint/default.nix:55]: pyproject.yaml says the opposite: https://github.com/adrienverge/yamllint/blob/b05e028c5881819161d11cb543fd96a30c06cceb/pyproject.toml#L6, so one of then has to be fixed.

Comment (NickCao) [reply pkgs/development/python-modules/yamllint/default.nix]: Opened PR upstream: https://github.com/adrienverge/yamllint/pull/580

**Pull request merged**